request-example.go

```diff
@@ -151,15 +170,38 @@ func (fs *root) Filecmd(r *Request) error {
 
 	switch r.Method {
 	case "Setstat":
+		// We explicitly do not support changing the mode of symlinks.
+		// Nor do we want to change the mode of whatever the symlink is pointed
+		// at.
+		link, err := fs.Readlink(r.Filepath)
+		if link != "" || err == nil || !errors.Is(err, os.ErrInvalid) {
+			return err
+		}
+
+		// Note: fs.openfile does not support opening a directory.
```

Comment:
Fortunately, this is all still passing the testing infrastructure, so I'm still very inclined to just keep permissions on directories as 'unimplemented' for right this moment.

Reply:
It’s passing the testing infrastructure merely because we’re not sufficiently testing it. :blobsweat:

I think if we’re going to start implementing permissions for files, but not for directories, then we’re going to need to start documenting what is covered and what is not. Right now attempting to change permissions on anything won’t do anything, and users are likely to say, “oh… I guess no permissions?” (Somewhat typical situation because of Windows.) Or changing ownership yields, “oh, I guess no ownership.”

But “wait, but file permissions work, and dir permissions don’t?!” is going to lead to questions that we can answer with documentation. If you’re not willing to put in the effort of :blobsweat: implementing the full permissions system. (I don’t blame you. I clearly stopped short.)

Reply:
The problem with implementing directory permissions is that I can't even remember the esoteric rules on what happens when one of read or execute is set without the other set, let alone the special bits.  Especially since we don't even have a chdir at that level, just get of specific full paths.  And I definitely recall that it gets weird when you have that be the case for an intermediate directory in the path, but you have full permissions for a subdirectory.

Only, now you've made me look at the POSIX specification with a migraine, ugh.

Writing test cases for this is going to be painful, but I'll take a look in my free time.  Sadly, having gotten stuff working 'well enough' for the work application, it's harder to spend work time on the remaining bits.